flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpoint.java

```diff
@@ -290,16 +289,14 @@ public static boolean checkpointsMatch(
 		Collection<CompletedCheckpoint> first,
 		Collection<CompletedCheckpoint> second) {
```

Comment:
change type to List? then we could skip the copy ado the comparison directly.

Reply:
I'm copying because I don't know what the exact type of the input will be and also because I need to extract the set of interesting fields explicitly. WDYT?

Reply:
Ah, I would've expected that CompletedCheckpoint has an equals method, never mind then.

Reply:
It did have, but that was removed a while ago because it was not well defined, so now I have to do it like this.